src/components/StripeCardInput/cardInput.js

```diff
@@ -0,0 +1,98 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { useTheme, useUniqueIdentifier } from '../../libs/hooks';
+import RenderIf from '../RenderIf';
+import Label from '../Input/label';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import StyledContainer from './styled/container';
+import StyledFakeDisabled from './styled/fakeDisabled';
+import getError from './helpers/getErrror';
+import getCardElementOptions from './helpers/getCardElementOptions';
+
+const CardInput = props => {
+    const {
+        label,
+        hideLabel,
+        bottomHelpText,
+        error,
+        className,
+        style,
+        disabled,
+        onChange,
+        onFocus,
+        onBlur,
+    } = props;
+    const stripe = useStripe();
+    const elements = useElements();
+    const cardElementId = useUniqueIdentifier('card-element');
+    const theme = useTheme().rainbow;
+    const cardElementOptions = useMemo(() => getCardElementOptions(theme, disabled), [
+        theme,
+        disabled,
+    ]);
+    const handleChange = event => {
+        if (stripe && elements) {
+            const card = {
+                stripe,
+                element: elements.getElement(CardElement),
+                iEmpty: event.empty,
```

Comment:
yes, typo

Reply:
How a consumer of this component it will use the object payload is being sent in the onChange callback

```js
const handleOnChange =  async (event) => {
     const { stripe, element } = event;
     ...
     const result = await stripe.confirmCardPayment('{CLIENT_SECRET}', {
          payment_method: {
              card: element,
              billing_details: {
                  name: 'Jenny Rosen',
              },
        }
    });
}

<StripeCardInput ... onChange={handleOnChange} />
```

my point the other day was to make the props in the event exactly like the props stripe use in their api then 

```js
const handleOnChange =  async (event) => {
     const { stripe, card } = event;
     ...
     const result = await stripe.confirmCardPayment('{CLIENT_SECRET}', {
          payment_method: {
              `card`,
              billing_details: {
                  name: 'Jenny Rosen',
              },
        }
    });
}
```

also `brand` prop could be changed to `cardBrand`

### suggested changes 

```js
const handleChange = event => {
        if (stripe && elements) {
            onChange({
                stripe,
                card: elements.getElement(CardElement),
                isEmpty: event.empty,
                isComplete: event.complete,
                cardBrand: event.brand,
                error: getError(event.error),
            };);
        }
    };
```